src/main/StartUp.java

```diff
@@ -1,7 +1,9 @@
 package main;
```

Comment:
Inside the method _runProgram_, the first line of code is: **headerLabel.setText(String.valueOf(array[charArrayIndex]));**. This should be moved inside _prepareGUI_.

Reply:
This line of code sets and displays the character that has to be typed. If  I move it to _prepareGUI_ I'll have to implement a method that is called every time a new character is displayed. Or do you have other suggestions?

Reply:
Yes, that would be a way.

I see now that _headerLabel_ property is called several times inside _runProgram_ method. You could leave the code as is here.